src/HackBuilder.php

```diff
@@ -181,13 +181,21 @@ public function openBrace(): this {
     return $this->addLine(' {')->indent();
   }
 
+  public function openBracket(): this {
+    return $this->addLine(' [')->indent();
```

Comment:
I see this is consistent with the existing ones, but we don't want the space for vec/dict/keyset. maybe `openBracket(ContainerType $type)` where ContainerType is an enum containing PHP_ARRAY, DICT, VEC, KEYSET ?

Reply:
How about `PHP_ARRAY`, `HACK_COLLECTION` and `HACK_ARRAY`?

Reply:
- hack collections don't use brackets
 - you need to insert one of ``, `vec`, `dict`, or `keyset` between the space and the `[`

Reply:
Oh, now I get what you mean I think.

My thought went into the direction of
```
public function getOpenClose(ContainerType $type): tuple(string, string) {
switch($type) {
case HACK_ARRAY: return ("[", "]");
case PHP_ARRAY: return (" [", "]");
case HACK_COLLECTION: return (" {", "}");
}
}
```
But your idea makes more sense :)